### ISSUE TITLE
installation instructions for v4 cause build fail with RN 0.68.0

### ISSUE THREAD

**Issue by infiniteline** [react-navigation/react-navigation#10759]
### Current behavior

- Create a new React Native 0.68.0 project
- Follow documentation to install ReactNavigation 4.4.4 and dependencies
- npx reat-native run-android
- build fails

This seems to have to do with the code in MainActivity.java. A 0.68.0 version app already has the following when created:
```
  @Override
  protected ReactActivityDelegate createReactActivityDelegate() {
    return new MainActivityDelegate(this, getMainComponentName());
  }

  public static class MainActivityDelegate extends ReactActivityDelegate {
    public MainActivityDelegate(ReactActivity activity, String mainComponentName) {
      super(activity, mainComponentName);
    }

    @Override
    protected ReactRootView createRootView() {
      ReactRootView reactRootView = new ReactRootView(getContext());
      // If you opted-in for the New Architecture, we enable the Fabric Renderer.
      reactRootView.setIsFabric(BuildConfig.IS_NEW_ARCHITECTURE_ENABLED);
      return reactRootView;
    }
  }
```

However, the installation instructions state that the following should be added:
```
  @Override
  protected ReactActivityDelegate createReactActivityDelegate() {
    return new ReactActivityDelegate(this, getMainComponentName()) {
      @Override
      protected ReactRootView createRootView() {
        return new RNGestureHandlerEnabledRootView(MainActivity.this);
      }
    };
  }
```

This results in a build fail as there are two ReactActivityDelegates. If you comment out the original createReactActivityDelegate call (from the bare app), then the app will build, but crash at start-up. This seems to be because the code in the bare app also creates a ReactRootView.

If you comment out the createRootView from the bare app, the build fails.

If you neglect to add in the code as specified in the ReactNavigation instructions, then the app will build and run. Navigation (once added) does work but will generate a Render error at some point (perhaps only with certain navigators?).

Not sure what versions of React Native this applies to. Previous project (where ReactNavigation was working) was 0.63.2.

### Expected behavior

Installation instructions allow the app to build and run without error.

### Reproduction

https://snack.expo.dev/@ericandrews/reactnavigation-v4---rn-0_68_0-build-fail?platform=android

### Platform

- [X] Android
- [ ] iOS
- [ ] Web

### Packages

- [X] react-navigation-​drawer
- [X] react-navigation-​stack
- [X] react-navigation-​tabs

### Environment

{
  "name": "AwesomeProject",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@react-native-masked-view/masked-view": "^0.2.7",
    "react": "17.0.2",
    "react-native": "0.68.0",
    "react-native-gesture-handler": "^2.5.0",
    "react-native-reanimated": "^2.9.1",
    "react-native-safe-area-context": "^4.3.1",
    "react-native-screens": "^3.15.0",
    "react-navigation": "^4.4.4",
    "react-navigation-stack": "^2.10.4",
    "react-navigation-tabs": "^2.11.2"
  },
  "devDependencies": {
    "@babel/core": "^7.18.10",
    "@babel/runtime": "^7.18.9",
    "@react-native-community/eslint-config": "^3.1.0",
    "babel-jest": "^28.1.3",
    "eslint": "^8.22.0",
    "jest": "^28.1.3",
    "metro-react-native-babel-preset": "^0.72.0",
    "react-test-renderer": "17.0.2"
  },
  "jest": {
    "preset": "react-native"
  }
}

**Comment by raajnadar** [comment]
Why are you using v4 why not v5 or v6?

**Comment by infiniteline** [comment]
This project was originally written with v4 and converting it to v5 is going to take a considerable amount of time. This may need to be done at some point, but we have a deadline to meet right now and (if at all possible) I'd rather update a few lines than invest in significant changes. If someone can tell me that v4 is absolutely incompatible with RN 0.68.0, then I'll have to take that route.

**Comment by raajnadar** [comment]
@infiniteline you can try using the https://reactnavigation.org/docs/5.x/compatibility/ and incrementally upgrade to v5

**Comment by infiniteline** [comment]
Yes, thanks @raajnadar. I had looked at this, but wasn't sure how well it would work due to some of the breaking changes. I'll try it and report back.

**Comment by jeremiebardon** [comment]
> Yes, thanks @raajnadar. I had looked at this, but wasn't sure how well it would work due to some of the breaking changes. I'll try it and report back.

Interested by how that work for you, I've tried without success at the moment !

**Comment by SusulAdam** [comment]
> > Yes, thanks @raajnadar. I had looked at this, but wasn't sure how well it would work due to some of the breaking changes. I'll try it and report back.
> 
> Interested by how that work for you, I've tried without success at the moment !

Have you managed to fix the problem? I would be very grateful for any tips on how to solve it 😄

**Comment by raajnadar** [comment]
The first step is to go through this upgrade document
https://reactnavigation.org/docs/5.x/upgrading-from-4.x

The second is to set up the compatibility layer
https://reactnavigation.org/docs/5.x/compatibility

In the third step, you need to check the getting started guide to confirm all the dependent versions & setup are correct
https://reactnavigation.org/docs/5.x/getting-started

Finally, you need to go through the document of the navigator to check the setup is correct, eg: - stack nav, drawer nav etc.
https://reactnavigation.org/docs/5.x/stack-navigator

Make sure you are reading the correct document for v5 check the URL containing `5.x`, first upgrade to v5 later you can start v6 which will be easier don’t jump to v6 directly from v4

Bonus - If you have an expo setup the upgrade will be very easier as the native setup for the dependant package will be handled by expo SDK.

**Comment by SusulAdam** [comment]
Due to that i working in the project which has few years and very very extensive navigation i tried omit upgrade of navigation. My version of navigation: 4.4.3

My steps to resolve problems: 
1. Add in dependencies: ` implementation 'androidx.work:work-runtime:2.7.1’` in app/build.gradle
2. If app still crash: upgrade "react-native-reanimated": "2.10.0" with necessary configuration. 
3. After upgrade reanimated I had couple problems with babel  to resolve it  I used command: `yarn upgrade metro-react-native-babel-preset` this caused that I was need install  `@babel/plugin-proposal-nullish-coalescing-operator `
4. go to node_modules/react-native-reanimated/RNReanimated.podspec 
5. yarn add patch-package
add script to the scripts at the package.json "postinstall": patch-package"
6. `Change "CLANG_CXX_LANGUAGE_STANDARD" => "c++14"`, on  `"CLANG_CXX_LANGUAGE_STANDARD" => "c++17"`, 
7. npx patch-package react-native-reanimated
